NoClassDefFoundError when using JAR as dependency

I built the JAR file using OpenJDK 8 and used it successfully as a standalone tool.

Now I want to use it as a dependency in a Java application that is built using Maven and OpenJDK 23.

I added the depedency to the pom.xml:

```
<dependency>
	<groupId>gedinline</groupId>
	<artifactId>gedinline.main</artifactId>
	<version>3.1.6</version>
	<scope>system</scope>
	<systemPath>${project.basedir}/libs/gedinline-3.1.6.jar</systemPath>
</dependency>
```

I made a minimal test call:

```
final var stringWriter = new StringWriter();
final var printWriter = new PrintWriter(stringWriter);

try {
	new GedInlineValidator(gedcom.getFile(), printWriter).validate();
	String result = stringWriter.toString();
	System.out.println(result);
} catch (FileNotFoundException e) {
	e.printStackTrace();
}
```

Then I get following errors during runtime:

```
java.lang.NoClassDefFoundError: Could not initialize class org.codehaus.groovy.vmplugin.v7.Java7
...
Caused by: java.lang.ExceptionInInitializerError: Exception org.codehaus.groovy.GroovyBugError [in thread "AWT-EventQueue-0"]
...
Exception in thread "AWT-EventQueue-0" java.lang.NoClassDefFoundError: Could not initialize class org.codehaus.groovy.reflection.ReflectionCache
...
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.NoClassDefFoundError: Could not initialize class org.codehaus.groovy.vmplugin.v7.Java7 [in thread "AWT-EventQueue-0"]
```

What can I do?

--- a/build.gradle
+++ b/build.gradle
@@ -42,7 +42,7 @@ task fatJar(type: Jar) {
         attributes 'Main-Class': 'gedinline.main.Main'
     }
 
-    archiveBaseName = 'gedinline'
+    archiveBaseName = 'gedinline-fat'
     duplicatesStrategy = DuplicatesStrategy.EXCLUDE
 
     from { configurations.runtimeClasspath.collect { it.isDirectory() ? it : zipTree(it) } }

--- a/build.gradle
+++ b/build.gradle
@@ -35,6 +35,10 @@ test {
     outputs.upToDateWhen { false }
 }
 
+jar {
+    archiveBaseName = 'gedinline-slim'
+}
+
 task fatJar(type: Jar) {
     dependsOn 'test'
 
@@ -42,7 +46,7 @@ task fatJar(type: Jar) {
         attributes 'Main-Class': 'gedinline.main.Main'
     }
 
-    archiveBaseName = 'gedinline-fat'
+    archiveBaseName = 'gedinline'
     duplicatesStrategy = DuplicatesStrategy.EXCLUDE
 
     from { configurations.runtimeClasspath.collect { it.isDirectory() ? it : zipTree(it) } }

